File: examples/go/cluster-list.go
package main

import (
	"context"
	"log/slog"
	"os"

	"connectrpc.com/connect"
	apiv1 "github.com/metal-stack-cloud/api/go/api/v1"
	"github.com/metal-stack-cloud/api/go/client"
)

func main() {

	token := os.Getenv("API_TOKEN")
	project := os.Getenv("PROJECT_ID")

	c := client.New(client.DialConfig{
		BaseURL: "https://api.metalstack.cloud",
		Token:   token,
	})

	ctx := context.Background()
	resp, err := c.Apiv1().Cluster().List(ctx, connect.NewRequest(&apiv1.ClusterServiceListRequest{
		Project: project,
	}))

	if err != nil {
		panic(err)
	}

	for _, cluster := range resp.Msg.Clusters {
		slog.Info("cluster", "name", cluster.Name, "id", cluster.Uuid, "kubernetes", cluster.Kubernetes.Version)
	}
}

File: examples/ts/cluster-list.ts
import { Interceptor, createPromiseClient } from "@connectrpc/connect";
import { ClusterService } from '@metal-stack-cloud/api/api/v1/cluster_connect'
import { createConnectTransport } from "@connectrpc/connect-node";
import dotenv from 'dotenv'; 
dotenv.config();

const tokenInterceptor: Interceptor = (next) => (req) => {
  req.header.append('Authorization', 'Bearer ' + process.env.API_TOKEN);
  return next(req);
}

const transport = createConnectTransport({
    interceptors: [tokenInterceptor],
    baseUrl: "https://api.metalstack.cloud",
    httpVersion: "1.1"
});


async function main() {
  const client = createPromiseClient(ClusterService, transport);
  const res = await client.list({ project: process.env.PROJECT_ID });

  for (let cluster of res.clusters) {
      console.log(cluster.name + " " + cluster.uuid);
  }
}
void main();